### PR TITLE
Two bug fixes for Mass2Zones and SplitFloors2ThermalZones

### PR DESCRIPTION
**Pull request by saeranv**
Two bug fixes here:

1. Mass2Zones - really small issue that was giving users workflow confusion referenced here: #666 . So I added a line of code to duplicate the input breps for the Mass2Zones so the output breps don't share a hbhive key with the input breps. 

2. SplitFloors2ThermalZones - Ji found an edge case (noted here #497) that was breaking this component. Essentially, square footprints were confusing the underlying straight skeleton algorithm, and producing degenerate polygons (2 lines).

Also an additional note. For the last little while I've been wondering why some of my commits to HB produce really weird git diffs like [this](https://github.com/mostaphaRoudsari/honeybee/compare/master...saeranv:master#diff-ec7d613ed740a8c7017d9d33f09aef43), where blank space looks like it's been deleted and re-added. And I think I finally isolated the problem: it only happens when I use the exportLadybug component. So for now, I'm going to avoid using this component for updating .py files, and just do it manually. It's not that much more work, and retains the utility of the git diff. 

-S

### CODE REVIEW

File: src/Honeybee_Masses2Zones.py [hunk 1]
```diff
@@ -85,22 +85,23 @@ def main(maximumRoofAngle, zoneMasses, zoneNames, zonePrograms, isConditioned):
...
     def duplicateData(data, calcLength):
         dupData = []
         for count in range(calcLength):
             dupData.append(data[0])
         return dupData
-    
+
     zoneNumber = len(zoneMasses)
-    
+    zoneMasses = map(lambda zm: copy.deepcopy(zm), zoneMasses) # Duplicate the breps so that we don't have same memory access
```
Comment (mostaphaRoudsari) [inline src/Honeybee_Masses2Zones.py]: @saeranv why do you use lambda? If you want to duplicate each item separately you can do something like this:
```python
zoneMasses = [copy.deepcopy(zm) for zm in zoneMasses]
```
BTW doesn't `zoneMasses = copy.deepcopy(zoneMasses)` work?

**Comment by saeranv** [comment]
@mostaphaRoudsari the main reason I use lambda is that's how we were taught at my school's CS class. I don't think there's any significant performance difference between map and list comprehension, however I know people like list comprehensions because it's clearer. 

One thing that I really like about lambdas is that it provides its own scope, while list comprehension may overwrite a variable. This is really helpful if your adding code to someone else's existing code i.e:

With lambda:
```
>> x = "a"
>> print map(lambda x: x, range(5))
[0,1,2,3,4]
>> print x
"a"
```
Without lambda:
```
>> x = "a"
>> print map(lambda x: x, range(5))
[0,1,2,3,4]
>>print x
4
```

Let me know if you'd like to change it to list comprehension, I'm not attached to either way! Also, I think you're right, deepcopy should work as well... perhaps that's the better solution. 

-S

**Comment by saeranv** [comment]
@mostaphaRoudsari Hmm there is a performance difference. I just checked [stackoverflow](https://stackoverflow.com/questions/36392224/python-map-or-list-comprehension) and the fastest way is to use map <i>without</i> the lambda! 

Assuming we want the fastest code, I'll revise this either with just map, or deepcopy depending on which is faster. Or would you prefer list comprehension?

**Comment by mostaphaRoudsari** [comment]
The will both work but as stated [here](https://stackoverflow.com/a/1247490/4394669): "Yep, indeed our internal Python style guide at work explicitly recommends listcomps against map and filter". Also [this](https://google.github.io/styleguide/pyguide.html?showone=Deprecated_Language_Features#Deprecated_Language_Features)

For the examples above try tuple( ) or ( ) that will be similar to lambda in having their own scope. I wasn't sure if we need indexing and/or multiple loops in this example otherwise using the generator is the most efficient way.

```python
x = "a"
create_generator = (x for x in range(5))
create_tuple = tuple(x for x in range(5))
print(x)
```

**Comment by saeranv** [comment]
Cool, I didn't know that tuples preserve scope. 

In this case, we do need to index, and append/extend the zoneMasses data structure, so I don't think tuples or generators will work. I always try to use generators whenever I can... but so far the only use case I've found is to enumerate through lists. 

I'll spend some time later today to revise the code and get rid of the map.

**Pull request closed without merging**